web-app/HeimScripts/heatmap/limmaUtils.R

```diff
@@ -33,3 +29,11 @@ getDesign <- function(measurements) {
 getSubset1Length <- function(measurements) {
   sum(grepl(pattern = SUBSET1REGEX, x = colnames(measurements))) # returns number of column names satisfying regexp.
 }
+
+# to check if row  contains at least one row that contains 3 non missing values
+rowContainsAtLeastThreeData <- function (row) {sum(!is.na(row)) > 2}
+
+# measurements should contains at least one valid row
+isValidLimmaMeasurements <- function (measurements) {
+   sum(apply(measurements, 1, rowContainsAtLeastThreeData)) > 1
```

Comment:
It is quite a lenient criterium - was limma crashing only when no rows had more than three cells filled? Or was it also crashing when there was at least one row with less than three cells filled?